src/context/WalletProvider/KeepKey/components/RecoverySettings.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  Button,
+  Flex,
+  FormControl,
+  FormLabel,
+  ModalBody,
+  ModalHeader,
+  Switch,
+  useColorModeValue,
+} from '@chakra-ui/react'
+import { useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router-dom'
+import { Radio, RadioOption } from 'components/Radio/Radio'
+import { Text } from 'components/Text'
+import { KeepKeyRoutes } from 'context/WalletProvider/routes'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+export const VALID_ENTROPY = [128, 192, 256] as const
+export type Entropy = typeof VALID_ENTROPY[number]
+
+export enum SentenceLength {
+  TwelveWords = VALID_ENTROPY[0],
+  EighteenWords = VALID_ENTROPY[1],
+  TwentyFourWords = VALID_ENTROPY[2],
+}
+
+const sentenceLengthOptions: readonly RadioOption<SentenceLength>[] = Object.freeze([
+  {
+    value: SentenceLength.TwelveWords,
+    label: ['modals.keepKey.recoverySettings.wordEntropy', { wordEntropy: '12' }],
+  },
+  {
+    value: SentenceLength.EighteenWords,
+    label: ['modals.keepKey.recoverySettings.wordEntropy', { wordEntropy: '18' }],
+  },
+  {
+    value: SentenceLength.TwentyFourWords,
+    label: ['modals.keepKey.recoverySettings.wordEntropy', { wordEntropy: '24' }],
+  },
+])
+
+export const KeepKeyRecoverySettings = () => {
+  const translate = useTranslate()
+  const history = useHistory()
+  const [useRecoveryPassphrase, setUseRecoveryPassphrase] = useState(false)
+  const [sentenceLengthSelection, setSentenceLengthSelection] = useState(
+    String(SentenceLength.TwelveWords),
+  )
+  const { setDeviceState } = useWallet()
+
+  const grayTextColor = useColorModeValue('gray.900', 'gray.400')
+  const grayBackgroundColor = useColorModeValue('gray.100', 'gray.700')
+
+  const isValidEntropy = (length: number | undefined): length is Entropy | undefined => {
+    return VALID_ENTROPY.some(e => e === length)
+  }
+
+  const handleSubmit = async () => {
+    const entropy = parseInt(sentenceLengthSelection)
+    const entropyTyped = isValidEntropy(entropy) ? entropy : VALID_ENTROPY[0]
```

Comment:
if we add `value: 12` to `sentenceLengthOption` and `entropy: 128` we can encapsulate all the necessary data and avoid this utility fn i think. cool type guard though

Reply:
Killed the type guard in https://github.com/shapeshift/web/pull/1561/commits/2c7fcf71bb69798876654abde57ab2d7eea13f1c

If we type `Entropy` as a string literal it gets more simple because `Radio` works only with strings, so we can drop the conversion and casting.